ldk/javascript/src/search/types.ts

```diff
@@ -0,0 +1,86 @@
+export enum FieldType {
+  /**
+   * Standard Filter is not case sensitive filters out english stop words
+   */
+  standard = 'standard',
+  /**
+   * Stemmer Filter ignores case but implements the porter stemmer and snowball stemmer filter
+   */
+  stemmer = 'stemmer',
+  /**
+   * Simple Filter only ignores case
+   */
+  simple = 'simple',
+  /**
+   * Numeric Filter is used to map numeric only fields
+   */
+  numeric = 'numeric',
+  /**
+   * Boolean Filter is used to map boolean or (1,0) only fields
+   */
+  boolean = 'boolean',
+  /**
+   * Datatime Filter is used to map datetime fields and support most major time standards
+   */
+  datetime = 'datetime',
+}
+
+export type Field = {
+  /**
+   * The name of the field
+   */
+  name: string;
+  /**
+   * The name of the field that will be used instead of the default name of the field (if provided)
+   */
+  displayName?: string;
+  /**
+   * The type of mapping for the field
+   */
+  type?: FieldType;
+};
+
+export type Document = {
+  /**
+   * The name of the document
+   */
+  name: string;
+  /**
+   * The data for the document that will be indexed and searched over
```

Comment:
Is there a specific format the data needs to be in in order to be indexed?

Reply:
> I'm not sure who should take responsibility for this, but given this is a new feature, and it's using a lot of different terminology within itself, it feels like our developer hub needs to be updated to give an explanation of how all these different parts interact.

To answer your first question, I have some written documentation but  I may need to pair with someone who has written some documentation for the LDK so it can be consistent with that is there

The data needs to be in a json like syntax, I have that in the documentation I have written but probably should add it in the comments here. It can accept valid json, typed object, or an array of typed objects that have been called with `JSON.stringify(...)`